### PR TITLE
query shelter with availibility (SDB-221)

### PR DESCRIPTION
**Pull request by Davit-BetterAngels**
SDB-221

## Summary by Sourcery

Add GraphQL support for querying shelters by bed availability and expose availability details on shelter types.

New Features:
- Expose shelter availability details on the shelter GraphQL type, including restricted and non-restricted beds.

Tests:
- Add GraphQL filter tests to verify shelters can be included or excluded based on whether they have available beds.

### CODE REVIEW

File: apps/betterangels-backend/shelters/tests/test_shelter_filters.py [hunk 1]
```diff
@@ -747,3 +747,71 @@ def test_shelter_open_now_excludes_partial_day_exception(self) -> None:
...
+        ShelterAvailability.objects.filter(shelter=shelter_with_non_restricted).update(
+            non_restricted_beds=5, restricted_beds=0
+        )
+        ShelterAvailability.objects.filter(shelter=shelter_with_restricted).update(
+            non_restricted_beds=0, restricted_beds=3
+        )
+        ShelterAvailability.objects.filter(shelter=shelter_no_available).update(
+            non_restricted_beds=0, restricted_beds=0
+        )
+        ShelterAvailability.objects.filter(shelter=shelter_zero_both).update(non_restricted_beds=0, restricted_beds=0)
```
Comment (tglaz) [inline apps/betterangels-backend/shelters/tests/test_shelter_filters.py]: is there a difference between the `shelter_no_available` and `shelter_zero_both` shelters?

File: apps/betterangels-backend/shelters/tests/test_shelter_filters.py [hunk 2]
```diff
@@ -747,3 +747,71 @@ def test_shelter_open_now_excludes_partial_day_exception(self) -> None:
...
+    def test_shelter_has_available_beds_filter_false(self) -> None:
+        """When hasAvailableBeds=false, only shelters WITHOUT available beds are returned."""
+        from shelters.models import ShelterAvailability
+
+        shelter_with_beds = shelter_recipe.make(status=StatusChoices.APPROVED)
+        shelter_no_beds = shelter_recipe.make(status=StatusChoices.APPROVED)
+
+        ShelterAvailability.objects.filter(shelter=shelter_with_beds).update(non_restricted_beds=2, restricted_beds=1)
+        ShelterAvailability.objects.filter(shelter=shelter_no_beds).update(non_restricted_beds=0, restricted_beds=0)
+
```
[the remark this answers is not in the document]
Comment (tglaz) [reply apps/betterangels-backend/shelters/tests/test_shelter_filters.py:797]: I'd suggest adding test for scenario where `hasAvailableBeds` value = `None` (returning all shelters).

for this feature I'm assuming that having the simple `hasAvailableBeds` boolean filter is sufficient.
i'd guess at some point we're able to filter by `non_restricted_beds` vs `restricted_beds` specifically - but doesn't seem specified in AC's

**Review by tglaz — CHANGES_REQUESTED** [review]

File: apps/betterangels-backend/shelters/tests/test_shelter_filters.py [hunk 3]
```diff
@@ -747,3 +747,71 @@ def test_shelter_open_now_excludes_partial_day_exception(self) -> None:
...
+        query = """
+            query ($filters: ShelterFilter) {
+                shelters(filters: $filters) {
+                    totalCount
+                    results { id }
+                }
+            }
+        """
+
+        response = self.execute_graphql(
```
Comment (tglaz) [inline apps/betterangels-backend/shelters/tests/test_shelter_filters.py:807]: may be good to test `expected_result_count` here also

**Review by tglaz — APPROVED** [review]
